app/services/inherited_proofing/va/field_validatable.rb

```diff
@@ -0,0 +1,78 @@
+module InheritedProofing
+  module Va
+    module FieldValidatable
```

Comment:
Is it feasible to use existing ActiveModel validations?  https://github.com/18F/identity-idp/blob/e116fa5fb036a94257aa3fd86b025033a4515478/app/forms/idv/doc_pii_form.rb is not perfect, but is a potential framework to work from.

Reply:
@mitchellhenke ty, I did look for existing code, but missed that model. That model assumes particular hash elements, assumes particular hash element key names and doesn't account for all the elements we're using; that doesn't mean we can't fudge key names and do overrides, but I think the benefit may not outweigh the cost. I thought about using ActiveModel, but AM seems too static and a bit more bulky than what we need right now. Perhaps I am missing something @holytoastr.

Reply:
There's some patterns to choose from in the forms in `app/forms`. Generally when doing validations in the IdP we use active model's validation DSL in a form and return a `FormResponse` object. There are some good positive downstream effects. For example, our analytics tooling is built to consume objects of the `FormResponse` class when logging the results of transactions.

Reply:
@gangelo Right, apologies, I wasn't intending to suggest re-using that, but rather taking the structure and fitting it for this.  Is there something ActiveModel is unable to accommodate for this use case?

Reply:
> @gangelo Right, apologies, I wasn't intending to suggest re-using that, but rather taking the structure and fitting it for this. Is there something ActiveModel is unable to accommodate for this use case?

No worries.

AM can do what we need but didn't see the need create an AM right now for a few reasons, although it may be worth a second look based the benefit of what hooper pointed out.

Reply:
> There's some patterns to choose from in the forms in `app/forms`. Generally when doing validations in the IdP we use active model's validation DSL in a form and return a `FormResponse` object. There are some good positive downstream effects. For example, our analytics tooling is built to consume objects of the `FormResponse` class when logging the results of transactions.

Will have a look.

Reply:
Zach has documented some of our patterns [here](https://github.com/18F/identity-idp/blob/main/docs/backend.md#forms-formresponse-analytics-and-controllers) and I think the preference is to stay within that.  We are relatively heavy users of ActiveModel for validation (both for things with or without database persistence), so a new validation module is less preferable.